internal/controller/nonadminrestore_controller.go

```diff
@@ -333,6 +333,28 @@ func (r *NonAdminRestoreReconciler) createVeleroRestore(ctx context.Context, log
 	}
 
 	if veleroRestore == nil {
+		if nar.Status.Phase == nacv1alpha1.NonAdminPhaseCreated {
```

Comment:
TODO test

I think I can not use phase here. `meta.IsStatusConditionTrue(nar.Status.Conditions, string(nacv1alpha1.NonAdminConditionQueued))` would be better

scenario I am afraid

NAR created -> (after restore operation finishes) velero restore deleted -> status is updated telling velero restore and also phase to BackingOff -> user updates NAR spec -> new velero restore is created